src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -284,18 +300,156 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
         // Only one buffer per state variable is required, we'll use the "output" one since this one captures the latest
         // tensor value
         allocate_tensor(stateName, stateDescriptor, TensorType::State, allocator);
+        _tensorsData[stateInputBufferName] = TensorData{_copyAllTensors.at(stateInputBufferName)->data(),
+                                                        _copyAllTensors.at(stateInputBufferName)->get_byte_size()};
+        _tensorsData[stateOutputBufferName] = TensorData{_copyAllTensors.at(stateOutputBufferName)->data(),
+                                                         _copyAllTensors.at(stateOutputBufferName)->get_byte_size()};
+    }
+}
+
+void ZeroInferRequest::check_and_get_mem_data(std::string name, bool isInput, bool isState) {
```

Comment:
This function looks complicated. Can you consider splitting it into one for remoteTensor case another one for a normal case?

Reply:
done!